File: frontend/lib/models/user_credentials.dart
class UserCredentials {
  final String ID;
  final String email;
  final String password;

  UserCredentials(
      {required this.ID,
        required this.email,
        required this.password});

  factory UserCredentials.fromJson(dynamic json) {
    return UserCredentials(
        ID: json['ID'] as String,
        email: json['email'] as String,
        password: json['password'] as String);
  }

  @override
  String toString() {
    return 'Recipe {ID: $ID, email: $email, password: $password}';
  }
}

File: backend/src/main/java/com/brainfood/security/Model/UserCredentials.java
package com.brainfood.security.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;
import org.springframework.stereotype.Component;

import java.io.Serializable;

@Component
@ToString
@Entity
@Table(name = "\"UserCredentials\"")
@Getter
@Setter
public class UserCredentials implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String ID;

    @Column(name = "username")
    private String username;

    @Column(name = "password")
    private String password;

    @Column(name = "email")
    private String email;

}
